apacheconfig/loader.py

```diff
@@ -326,8 +339,14 @@ def loads(self, text, initialize=True, source=None):
             text (Text): (Text) containing the configuration to load.
 
         Returns:
-            (dict) containing configuration information loaded from text.
+            (dict) containing configuration information loaded from text; if
+            `writable` is set on this object, returns a mutable
+            :class:`apacheconfig.ListNode` AST containing parsed config file.
         """
+        if self._writable:
+            ast = self._parser.parse(text)
+            return ListNode(ast, self._parser)
```

Comment:
I fear that making return type dependent on method parameter can be confusing... And it probably does not save us from prospective deprecation.

What I had in mind is to ditch the entire original `ApacheConfigLoader`'s body leaving just `loads()` for compatibility:

    def loads(self, text, initialize=True, source=None):
        ast = self._parser.parse(text)
        return ListNode(ast, self._parser).as_dict()

But that requires `as_dict()` that walks the entire AST tree (which is already in form of loader objects) and produces Python types from each object. I am assuming (may be wrongly?), that turning wloader objects into Python types is cheaper than walking the original AST as `ApacheConfigLoader` does...

WDYT?

I understand that it's a larger undertaking, so may be it's still better to make a two-loader release as still requiring deprecation, but arguably less confusing...?

Thoughts?

Reply:
Yes, I agree that this behavior is strange. I think it might be better to leave them separate-- and slowly migrate the `as_dict` functionality into `WritableLoader`. It will likely take some time-- I'm not sure we can avoid the deprecation concern so long as the two Loaders return separate AST objects.

Reply:
So are we good to cut a release?

Reply:
I think so! :)

Reply:
#93